src/core_modules/capture-core/components/Pages/EnrollmentEvent/EnrollmentEventPage.component.js

```diff
@@ -23,7 +35,17 @@ const EnrollmentEventPagePain = ({ mode, classes }) => (
                     escape: ':',
                 })}
         </div>
-    </>
+        {programStage ? (
```

Comment:
In case we don't find `programStage` this will be loading forever? I would suggest separating the state of loading when waiting for results in program vs the state we don't find `programStage` match `stageId`. WDYT?